src/tests/account/test_secp256r1.cairo

```diff
@@ -0,0 +1,141 @@
+
+    assert_eq!(x, private_key);
+    assert_eq!(parity, false, "Parity should be even");
+}
+
+#[test]
+fn test_unpack_big_secp256r1_points() {
+    let (big_point_1, big_point_2) = get_points();
+
+    // Check point 1
+
+    let (expected_x, expected_y) = big_point_1.get_coordinates().unwrap_syscall();
+
+    let (xlow, xhigh_and_parity) = StorePacking::pack(big_point_1);
+    let (x, y) = StorePacking::unpack((xlow, xhigh_and_parity)).get_coordinates().unwrap_syscall();
+
+    assert_eq!(x, expected_x);
+    assert_eq!(y, expected_y);
+
+    // Check point 2
+
+    let (expected_x, _) = big_point_2.get_coordinates().unwrap_syscall();
+
+    let (xlow, xhigh_and_parity) = StorePacking::pack(big_point_2);
+    let (x, _) = StorePacking::unpack((xlow, xhigh_and_parity)).get_coordinates().unwrap_syscall();
+
+    assert_eq!(x, expected_x);
+    assert_eq!(y, expected_y);
+}
+
+#[test]
+fn test_secp256r1_serialization() {
+    let (big_point_1, big_point_2) = get_points();
+
+    let mut serialized_point = array![];
+    let mut expected_serialization = array![];
+
+    // Check point 1
+
+    big_point_1.serialize(ref serialized_point);
+    big_point_1.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+
+    assert!(serialized_point == expected_serialization);
+
+    // Check point 2
+
+    big_point_2.serialize(ref serialized_point);
+    big_point_2.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+
+    assert!(serialized_point == expected_serialization);
+}
+
+#[test]
+fn test_secp256r1_deserialization() {
+    let (big_point_1, big_point_2) = get_points();
+
+    // Check point 1
+
+    let mut expected_serialization = array![];
+
+    big_point_1.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+    let mut expected_serialization = expected_serialization.span();
+    let deserialized_point = Secp256r1PointSerde::deserialize(ref expected_serialization).unwrap();
+
+    assert_eq!(big_point_1, deserialized_point);
+
+    // Check point 2
+
+    let mut expected_serialization = array![];
+
+    big_point_2.get_coordinates().unwrap_syscall().serialize(ref expected_serialization);
+    let mut expected_serialization = expected_serialization.span();
+    let deserialized_point = Secp256r1PointSerde::deserialize(ref expected_serialization).unwrap();
+
+    assert_eq!(big_point_2, deserialized_point);
+}
+
+#[test]
+fn test_partial_eq() {
+    let (big_point_1, big_point_2) = get_points();
+
+    assert_eq!(big_point_1, big_point_1);
+    assert_eq!(big_point_2, big_point_2);
+    assert!(big_point_1 != big_point_2);
+    assert!(big_point_2 != big_point_1);
+}
+
+//
+// Helpers
+//
+
+/// This signature was computed using @noble/curves.
+fn P256_PRIVATEKEY_SAMPLE() -> u256 {
+    0x1efecf7ee1e25bb87098baf2aaab0406167aae0d5ea9ba0d31404bf01886bd0e
+}
+
+fn get_points() -> (Secp256r1Point, Secp256r1Point) {
+    let private_key = P256_PRIVATEKEY_SAMPLE();
```

Comment:
Do we need x to be a private key? The name seems a bit misleading.

Reply:
I can replace the x value on the curve by a const and remove the function. Whatever you want but tell me what you prefer otherwise we may spin:
- do you want a const or simply the value?
- if you prefer a const or a function, how would you like I name it?

Reply:
In this case, I think the value is okay.